### PR TITLE
make SGX as part of release-1.16

### PR DESCRIPTION
**Pull request by zhlsunshine**
**Please provide a description of this PR:**
make SGX as part of release-1.16
1. Added bootstrap configuration for SGX
2. Updated the ingress gateway's deployment for SGX

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by poussa** [comment]
@zhlsunshine are you making SGX mandatory for all ingress gateway deployments. That is not we want. This needs to be opt-in.

File: manifests/charts/gateways/istio-ingress/templates/deployment.yaml [hunk 1]
```diff
@@ -266,8 +273,18 @@ spec:
             readOnly: true
           {{- end }}
           {{- end }}
-{{- if $gateway.additionalContainers }}
-{{ toYaml $gateway.additionalContainers | indent 8 }}
+{{- $noSGX := true }}
+{{- range $container := $gateway.additionalContainers }}
+{{ if and ($noSGX) (contains "sds" $container.name) }}
```
Comment (ipuustin) [inline manifests/charts/gateways/istio-ingress/templates/deployment.yaml:278]: Hmm, is the logic wrong way around? I mean, should it be (and (not noSGX) ....) to enable SGX?

File: manifests/charts/gateways/istio-ingress/templates/deployment.yaml [hunk 2]
```diff
@@ -266,8 +273,18 @@ spec:
             readOnly: true
           {{- end }}
           {{- end }}
-{{- if $gateway.additionalContainers }}
-{{ toYaml $gateway.additionalContainers | indent 8 }}
+{{- $noSGX := true }}
```
Comment (poussa) [inline manifests/charts/gateways/istio-ingress/templates/deployment.yaml:276]: Where is this noSGX variable coming from?

Comment (zhlsunshine) [reply manifests/charts/gateways/istio-ingress/templates/deployment.yaml:276]: Hi @poussa, L#276 is the source of `noSGX` variable.

File: manifests/charts/gateways/istio-ingress/templates/deployment.yaml (hunk 1 above)
Comment (zhlsunshine) [reply manifests/charts/gateways/istio-ingress/templates/deployment.yaml:278]: Hi @ipuustin, in fact I may need to change the variable name as `enableSGX`

**Comment by zhlsunshine** [comment]
Hi @ipuustin & @poussa, resubmitted, please help to review, thanks!

**Review by irisdingbj — APPROVED** [review]
thanks!

**Pull request merged**